src/track/trackref.cpp

```diff
@@ -0,0 +1,43 @@
+#include "track/trackref.h"
+
+
+bool TrackRef::verifyConsistency() const {
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasCanonicalLocation()) {
+        return false;
+    }
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasId()) {
+        return false;
+    }
+    return true;
+}
+
+QFileInfo TrackRef::createFileInfo() const {
+    QFileInfo fileInfo(getLocation());
```

Comment:
I'm already working on that ;)

TrackInfoObject gets back an instance of QFileInfo and createFileInfo() will 
only be invoked during tests and maybe in DEBUG_ASSERTs.

On 09/17/2015 03:44 PM, Daniel Schürmann wrote:
>
> In src/track/trackref.cpp 
> <https://github.com/mixxxdj/mixxx/pull/709#discussion_r39746184>:
>
> > @@ -0,0 +1,43 @@
> > +#include "track/trackref.h"
> > +
> > +
> > +bool TrackRef::verifyConsistency() const {
> > +    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasCanonicalLocation()) {
> > +        return false;
> > +    }
> > +    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasId()) {
> > +        return false;
> > +    }
> > +    return true;
> > +}
> > +
> > +QFileInfo TrackRef::createFileInfo() const {
> > +    QFileInfo fileInfo(getLocation());
>
> I probably have not understood the whole picture this PR is heading for, 
> sorry for my critics in advanced.
>
> You have introduces the TrackRef to encapsulates all identifying 
> properties of TrackInfoObject.
> This reduces indeed the complexity from the TrackInfo object, but 
> currently clutters the using code and may introduce a performance regression.
>
> What will be possible after this PR is applied, what we can't do now?
> The lock less access of the location seams to be a win.
>
> Your approach would work, if we don't have to deal with system calls via 
> QFileInfo.
> Can't we reintroduce a private class inside TrackInfoObject, that handles 
> TrackRef and a locked QFileInfo?
>
> In current Master, we have a single managed TrackPointer, that caries all 
> track related info and is accessible via a common interface, that is not 
> that bad. What are the issues.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/709/files#r39746184>.
>